File: doc/src/codes/Users_Manual/gain/holo_1.py
from pyautd3 import EmitIntensity
from pyautd3.gain.holo import GSPAT, EmissionConstraint, NalgebraBackend

backend = NalgebraBackend()
g = GSPAT(backend).with_constraint(EmissionConstraint.uniform(EmitIntensity.maximum()))

File: doc/src/codes/Users_Manual/gain/holo_1.cpp
#include "autd3/gain/holo.hpp"

const auto backend = std::make_shared<autd3::gain::holo::NalgebraBackend>();
auto g = autd3::gain::holo::GSPAT(backend).with_constraint(
    autd3::gain::holo::EmissionConstraint::uniform(
        autd3::EmitIntensity::maximum()));